pkg/manager/daemon_adaptor.go

```diff
@@ -192,6 +195,19 @@ func (m *Manager) BuildDaemonCommand(d *daemon.Daemon, bin string, upgrade bool)
 
 	cmd := exec.Command(nydusdPath, args...)
 
+	if config.IsKeyringEnabled() && !config.IsFusedevSharedModeEnabled() {
+		c, err := daemonconfig.NewDaemonConfig(d.States.FsDriver, configPath)
```

Comment:
I am afraid the current way to identify a container image reference's host is not agile.  For the dedicated mode nydusd daemon, it only as a rafs instance in its Instances where we can calculate what host is

Reply:
Fixed.